esp/esp/users/models/__init__.py

```diff
@@ -920,9 +917,23 @@ def current_schoolyear(program=None):
             schoolyear = curyear + 1
         return schoolyear
 
+    @staticmethod
     @cache_function
-    def getGrade(self, program = None):
-        grade = 0
+    def program_schoolyear(program):
+        schoolyear = Tag.getProgramTag('schoolyear', program)
+        if schoolyear is not None:
+            # tag override
+            schoolyear = int(schoolyear)
+        else:
+            # "now" is actually whenever the program ran or will run
+            now = program.dates()[0]
+            schoolyear = ESPUser.current_schoolyear(now)
+        schoolyear += program.incrementGrade() # adds 1 if appropriate tag is set; else does nothing
+        return schoolyear
+    program_schoolyear.__func__.depend_on_row(lambda: Tag, lambda tag: {'program': tag.target})
+
+    @cache_function
+    def getYOG(self, program = None):
```

Comment:
Can you add a docstring here?

When I was reading the code changes above, I was confused why ``getYOG`` would take a ``program`` argument. It's clear from looking at the implementation, but for clarity, the behavior should be documented.